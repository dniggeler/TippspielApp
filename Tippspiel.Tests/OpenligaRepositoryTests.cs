using FluentAssertions;
using Ninject;
using System.Threading.Tasks;
using BhFS.Tippspiel.Utils;
using Snapshooter.Xunit;
using Tippspiel.Contracts;
using Xunit;

namespace Tippspiel.Tests
{
    [Trait("Tippspiel","Openliga Repository")]
    public class OpenligaRepositoryTests
    {
        private readonly IFussballDataRepository _repository;

        public OpenligaRepositoryTests()
        {
            StandardKernel kernel = new StandardKernel( new TippspielModule() );
            _repository = kernel.Get<IFussballDataRepository>();
        }

        [Fact(DisplayName = "Get Matches By Group")]
        public async Task GetMatchesByGroupTest()
        {
            // given
            int groupId = 3;

            // when
            var result = await _repository.GetMatchesByGroupAsync(groupId);

            // then
            Snapshot.Match(result);
        }

        [Fact(DisplayName = "Get Current Group")]
        public async Task GetCurrentGroupTest()
        {
            // given

            // when
            var result = await _repository.GetCurrentGroupAsync();

            // then
            Snapshot.Match(result);
        }

        [Fact(DisplayName = "Get All Groups")]
        public async Task GetAllGroupsTest()
        {
            // given
            int expectedMatches = 34;

            // when
            var groups = await _repository.GetAllGroupsAsync();

            // then
            groups.Count.Should().Be(expectedMatches);
        }

        [Fact(DisplayName = "Get All Matches")]
        public async Task GetAllMatchesTest()
        {
            // given
            int expectedMatches = 306;

            // when
            var matches = await _repository.GetAllMatchesAsync();

            // then
            matches.Count.Should().Be(expectedMatches);
        }


        [Fact(DisplayName = "Get Match By Id")]
        public async Task GetMatchById()
        {
            // given
            int matchId = 66641;

            // when
            var match = await _repository.GetMatchDataAsync(matchId);

            // then
            Snapshot.Match(match);
        }

        [Fact(DisplayName = "Get Match Not Started")]
        public async Task GetMatchByIdNotYetStarted()
        {
            // given
            int matchId = 66648;

            // when
            var match = await _repository.GetMatchDataAsync(matchId);

            // then
            Snapshot.Match(match);
        }

        [Fact(DisplayName = "Get Teams")]
        public async Task GetTeamsTest()
        {
            // given
 
            // when
            var result = await _repository.GetTeamsAsync();

            // then
            Snapshot.Match(result);
        }

        [Fact(DisplayName = "Get Next Match")]
        public async Task GetNextMatch()
        {
            // given

            // when
            var result = await _repository.GetNextMatchAsync();

            // then
            Snapshot.Match(result);
        }

        [Fact(DisplayName = "Get Last Match")]
        public async Task GetLastMatch()
        {
            // given

            // when
            var result = await _repository.GetLastMatchAsync();

            // then
            Snapshot.Match(result);
        }
    }
}
