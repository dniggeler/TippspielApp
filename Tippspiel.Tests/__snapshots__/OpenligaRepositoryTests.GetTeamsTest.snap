[
  {
    "teamId": 199,
    "teamName": "1. FC Heidenheim 1846",
    "shortName": "Heidenheim",
    "teamIconUrl": "https://upload.wikimedia.org/wikipedia/commons/thumb/9/9d/1._FC_Heidenheim_1846.svg/830px-1._FC_Heidenheim_1846.svg.png"
  },
  {
    "teamId": 65,
    "teamName": "1. FC Köln",
    "shortName": "Köln",
    "teamIconUrl": "https://upload.wikimedia.org/wikipedia/commons/thumb/0/01/1._FC_Koeln_Logo_2014–.svg/296px-1._FC_Koeln_Logo_2014–.svg.png"
  },
  {
    "teamId": 80,
    "teamName": "1. FC Union Berlin",
    "shortName": "Union Berlin",
    "teamIconUrl": "https://assets.dfb.de/uploads/000/018/232/small_union-Berlin.jpg"
  },
  {
    "teamId": 81,
    "teamName": "1. FSV Mainz 05",
    "shortName": "Mainz",
    "teamIconUrl": "https://upload.wikimedia.org/wikipedia/commons/thumb/9/9e/Logo_Mainz_05.svg/1200px-Logo_Mainz_05.svg.png"
  },
  {
    "teamId": 6,
    "teamName": "Bayer Leverkusen",
    "shortName": "Leverkusen",
    "teamIconUrl": "https://upload.wikimedia.org/wikipedia/de/thumb/f/f7/Bayer_Leverkusen_Logo.svg/1200px-Bayer_Leverkusen_Logo.svg.png"
  },
  {
    "teamId": 7,
    "teamName": "Borussia Dortmund",
    "shortName": "BVB",
    "teamIconUrl": "https://upload.wikimedia.org/wikipedia/commons/thumb/6/67/Borussia_Dortmund_logo.svg/560px-Borussia_Dortmund_logo.svg.png"
  },
  {
    "teamId": 87,
    "teamName": "Borussia Mönchengladbach",
    "shortName": "Gladbach",
    "teamIconUrl": "https://i.imgur.com/KSIk0Eu.png"
  },
  {
    "teamId": 91,
    "teamName": "Eintracht Frankfurt",
    "shortName": "Frankfurt",
    "teamIconUrl": "https://i.imgur.com/X8NFkOb.png"
  },
  {
    "teamId": 95,
    "teamName": "FC Augsburg",
    "shortName": "Augsburg",
    "teamIconUrl": "https://i.imgur.com/sdE62e2.png"
  },
  {
    "teamId": 40,
    "teamName": "FC Bayern München",
    "shortName": "Bayern",
    "teamIconUrl": "https://i.imgur.com/jJEsJrj.png"
  },
  {
    "teamId": 1635,
    "teamName": "RB Leipzig",
    "shortName": "Leipzig",
    "teamIconUrl": "https://i.imgur.com/Rpwsjz1.png"
  },
  {
    "teamId": 112,
    "teamName": "SC Freiburg",
    "shortName": "Freiburg",
    "teamIconUrl": "https://i.imgur.com/r3mvi0h.png"
  },
  {
    "teamId": 118,
    "teamName": "SV Darmstadt 98",
    "shortName": "Darmstadt",
    "teamIconUrl": "https://upload.wikimedia.org/wikipedia/commons/e/e5/SV_Darmstadt_98_Logo.svg"
  },
  {
    "teamId": 175,
    "teamName": "TSG 1899 Hoffenheim",
    "shortName": "Hoffenheim",
    "teamIconUrl": "https://i.imgur.com/gF0PfEl.png"
  },
  {
    "teamId": 16,
    "teamName": "VfB Stuttgart",
    "shortName": "Stuttgart",
    "teamIconUrl": "https://i.imgur.com/v0tkpNx.png"
  },
  {
    "teamId": 129,
    "teamName": "VfL Bochum",
    "shortName": "Bochum",
    "teamIconUrl": "https://i.imgur.com/5jy3Gfr.png"
  },
  {
    "teamId": 131,
    "teamName": "VfL Wolfsburg",
    "shortName": "Wolfsburg",
    "teamIconUrl": "https://i.imgur.com/ucqKV4B.png"
  },
  {
    "teamId": 134,
    "teamName": "Werder Bremen",
    "shortName": "Bremen",
    "teamIconUrl": "https://upload.wikimedia.org/wikipedia/commons/thumb/b/be/SV-Werder-Bremen-Logo.svg/681px-SV-Werder-Bremen-Logo.svg.png"
  }
]
